builders/xcm/fees.md

```diff
@@ -0,0 +1,230 @@
+
+Now that you are aware of the weight costs for database reads and writes on Kusama, you can calculate the weight cost for a given instruction using the base weight for an instruction. 
+
+The `WithdrawAsset` instruction has a base weight of `20,385,000`, and includes one database read, and one database write. Therefore, the total weight cost of the `WithdrawAsset` instruction is calculated like so:
+
+```
+20,385,000 + 25,000,000 + 100,000,000 = 145,385,000
+```
+
+The `BuyExecution` instruction has a base weight of `3,109,000` and doesn't include any database reads or writes. Therefore, the total weight cost of the `BuyExecution` instruction is `3,109,000`.
+
+With the instruction weight cost established, you can calculate the cost of the instruction in KSM. 
+
+In Kusama, the `ExtrinsicBaseWeight` is set to `86,309,000` which is mapped to 1/10 cents. Where 1 cent is `10^12 / 30,000`. Therefore, the formula for calculating the final fee in KSM looks like this:
+
+```
+Total-Planck-KSM = TotalWeight * PlanckKSM * (1 / ExtrinsicBaseWeight)
+KSM = Total-Planck-KSM / 10^12
+```
+
+Therefore, the actual calculation for the `WithdrawAsset` instruction is:
+
+```
+Total-Planck-KSM = 145,385,000 * (10^12 / 30000 * 10) * (1 / 86309000)
+KSM = 5614903.04216 / 10^12
+```
+
+The total cost is `0.00000561490304213694 KSM`.
+
+### Moonbeam-based Networks {: #moonbeam-based-networks }
+
+Moonbeam uses a fixed amount of weight for each XCM instruction. The amount for each of the Moonbeam-based networks is as follows:
+
+=== "Moonbeam"
+    ```
+    200,000,000 weight units
+    ```
+
+=== "Moonriver"
+    ```
+    200,000,000 weight units
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    100,000,000 weight units
+    ```
+
+Weight units are converted to balance units as part of the fee calculation. For XCM related weight, the cost is fixed for each Moonbeam-based network as follows:
+
+=== "Moonbeam"
+    ```
+    5,000,000 Wei per weight
+    ```
+
+=== "Moonriver"
+    ```
+    50,000 Wei per weight
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    50,000 Wei per weight
+    ```
+
+This means that on Moonbase Alpha, for example, the formula to calculate the cost of one XCM instruction is as follows:
+
+```
+Wei = Weight * Wei_Per_Weight
+DEV = Wei / (10^18)
+```
+
+Therefore, the actual calculation is:
+
+```
+Wei = 5000000 * 200000000
+DEV = 5000000000000 / (10^18)
+```
+
+The total cost is `0.000005 DEV` for an XCM instruction on Moonbase Alpha.
+
+## Calculation of Fees for External Assets on Moonbeam {: #calculation-of-fees-external-asset }
+
+In the previous example with Alice, her fees were taken out of Alith's xcDOT balance. To determine the amount of a particular asset, such as xcDOT, to charge per second of local XCM execution, Moonbeam uses a concept called `UnitsPerSecond`. This concept is used by parachains to determine how much to charge in the target parachain. Nevertheless, fees can be charged in another token, for example, DOT.
```

Comment:
I would continue using the Alith example.

"To determined the amount Alith was charged in xcDOT, Moonbeam uses the concept called `UnitsPerSecond`, which refers to units of tokens that the network charges per second of XCM execution time...."